emission/analysis/section_features.py

```diff
@@ -15,18 +15,40 @@
 
 
 # The speed is in m/s
-def calSpeed(trackpoint1, trackpoint2):
+def calSpeed(section):
   from dateutil import parser
-  distanceDelta = calDistance(trackpoint1['track_location']['coordinates'],
-                              trackpoint2['track_location']['coordinates'])
-  timeDelta = parser.parse(trackpoint2['time']) - parser.parse(trackpoint1['time'])
+
+  if ((not section['start_loc']) or (not section['end_loc'])):
```

Comment:
Also this appears to be the speed for the entire section instead of between every pair of points along the trajectory.